### PR TITLE
cloudflare-utils: init at 1.2.1

### PR DESCRIPTION
**Pull request by yelite**
## Description of changes

Add cloudflare-utils https://github.com/Cyb3r-Jak3/cloudflare-utils, Helpful Cloudflare utility program 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/cloudflare-utils/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "Cyb3r-Jak3";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-41TQ+St6U4exLSl4dwc1E6K8P+oqQ4m5RSI7L2/dWwI=";
+  };
+
+  vendorHash = "sha256-HE6x4KSe9b9ZzcYz7sP25aTeDGU4zXgkYm/1RwYYBt4=";
+
+  meta = with lib; {
```
Comment (doronbehar) [inline pkgs/by-name/cl/cloudflare-utils/package.nix:19]: Many people claim that using `with lib;` in such a wide scope is not good. I tend to agree with this point.

File: pkgs/by-name/cl/cloudflare-utils/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    description = "Helpful Cloudflare utility program";
+    homepage = "https://github.com/Cyb3r-Jak3/cloudflare-utils";
+    license = licenses.asl20;
```
Comment (doronbehar) [inline pkgs/by-name/cl/cloudflare-utils/package.nix:22]: You could use `lib.licences.ask20` here.

File: pkgs/by-name/cl/cloudflare-utils/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    maintainers = with maintainers; [ yelite ];
```
Comment (doronbehar) [inline pkgs/by-name/cl/cloudflare-utils/package.nix:23]: And do use `with lib.maintainers;` here.

File: pkgs/by-name/cl/cloudflare-utils/package.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule rec {
+  pname = "cloudflare-utils";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "Cyb3r-Jak3";
+    repo = pname;
```
Comment (doronbehar) [inline pkgs/by-name/cl/cloudflare-utils/package.nix:12]: There's also a common argument against using `repo = pname` is that it creates a not really needed link between the two attributes.

**Review by doronbehar — COMMENTED** [review]
A few comments, looks good overall.

File: pkgs/by-name/cl/cloudflare-utils/package.nix (hunk 1 above)
Comment (yelite) [reply pkgs/by-name/cl/cloudflare-utils/package.nix]: This makes sense. I removed `with lib;`

File: pkgs/by-name/cl/cloudflare-utils/package.nix (hunk 4 above)
Comment (yelite) [reply pkgs/by-name/cl/cloudflare-utils/package.nix]: Updated

**Pull request merged**